server/src/main/java/com/qubole/quark/serverclient/server/Main.java

```diff
@@ -0,0 +1,164 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.qubole.quark.serverclient.server;
+
+import org.apache.calcite.avatica.Meta;
+import org.apache.calcite.avatica.remote.Driver;
+import org.apache.calcite.avatica.remote.LocalService;
+import org.apache.calcite.avatica.remote.Service;
+import org.apache.calcite.avatica.server.HandlerFactory;
+import org.apache.calcite.avatica.server.HttpServer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.eclipse.jetty.server.Handler;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
+import java.util.Collections;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A query server for Phoenix over Calcite's Avatica.
+ */
+public final class Main implements Runnable {
+
+  protected static final Log LOG = LogFactory.getLog(Main.class);
+
+  private final String[] argv;
+  private final CountDownLatch runningLatch = new CountDownLatch(1);
+  private HttpServer server = null;
+  private int retCode = 0;
+  private Throwable t = null;
+
+  /**
+   * Log information about the currently running JVM.
+   */
+  public static void logJVMInfo() {
+    // Print out vm stats before starting up.
+    RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+    if (runtime != null) {
+      LOG.info("vmName=" + runtime.getVmName() + ", vmVendor="
+          + runtime.getVmVendor() + ", vmVersion=" + runtime.getVmVersion());
+      LOG.info("vmInputArguments=" + runtime.getInputArguments());
+    }
+  }
+
+  public Main() {
+    this(null);
+  }
+
+  public Main(String[] argv) {
+    this.argv = argv;
+  }
+
+  /**
+   * @return the port number this instance is bound to, or {@code -1} if the server is not running.
+   */
+  public int getPort() {
+    if (server == null) {
+      return -1;
+    }
+    return server.getPort();
+  }
+
+  /**
+   * @return the return code from running as a.
+   */
+  public int getRetCode() {
+    return retCode;
+  }
+
+  /**
+   * @return the throwable from an unsuccessful run, or null otherwise.
+   */
+  public Throwable getThrowable() {
+    return t;
+  }
+
+  /** Calling thread waits until the server is running. */
+  public void awaitRunning() throws InterruptedException {
+    runningLatch.await();
+  }
+
+  /** Calling thread waits until the server is running. */
+  public void awaitRunning(long timeout, TimeUnit unit) throws InterruptedException {
+    runningLatch.await(timeout, unit);
+  }
+
+  public void run(String[] args) throws Exception {
+    logJVMInfo();
+    try {
+
+      Class<? extends Meta.Factory> factoryClass = QuarkMetaFactoryImpl.class;
+      int port = 8765;
```

Comment:
Make this configurable (whenever you make changes to pass relevant arguments to server)